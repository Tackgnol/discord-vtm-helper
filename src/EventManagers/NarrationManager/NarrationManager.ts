import { IReply, ReplyType } from '../../Models/AppModels';
import { narrationRichEmbed } from '../../Common';
import { InvalidInputError } from '../../Common/Errors';

class NarrationManager {
	displayNarration(narrationText: string, image?: string): IReply {
		if (narrationText.length === 0) {
			throw new InvalidInputError('No text narration to display');
		}
		return { type: ReplyType.Channel, value: narrationRichEmbed(narrationText, image) };
	}
}

export default NarrationManager;
