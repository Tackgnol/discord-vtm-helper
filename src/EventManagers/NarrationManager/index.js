const TurndownService = require('turndown');

class NarrationManager {
	constructor(message, channel) {
		this.message = message;
		this.channel = channel;
	}

	displayNarration(narrationText, image = null) {
		const messageChanel = this.channel;
		const turndownService = new TurndownService()
		const markdown = turndownService.turndown(narrationText)
		messageChanel.send(`${markdown} \n ${image}`);
	}

}

module.exports = NarrationManager;
