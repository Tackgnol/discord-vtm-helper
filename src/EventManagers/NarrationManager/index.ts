import { narrationRichEmbed } from '../../Common/RichEmbeds';
import { Message, TextChannel } from 'discord.js';

class NarrationManager {
	constructor(private channel: TextChannel, private message?: Message) {
		this.message = message;
		this.channel = channel;
	}

	displayNarration(narrationText: string, image?: string) {
		const messageChanel = this.channel;
		const narration = narrationRichEmbed(narrationText, image);
		messageChanel.send(narration);
	}
}

export default NarrationManager;
