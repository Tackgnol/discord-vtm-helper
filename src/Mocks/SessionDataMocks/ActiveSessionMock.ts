import { IActiveSession } from '../../Models/AppModels';
import { existingGTQueryWithoutValue } from '../QueryMocks/globalTestQueries';

export const activeSessionMock: IActiveSession = {
	channelId: '0',
	gameId: '0',
	prevCommand: existingGTQueryWithoutValue,
};
