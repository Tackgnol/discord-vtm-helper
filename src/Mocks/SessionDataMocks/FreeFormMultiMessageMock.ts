export const freeFormMultiMessageMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['2', '3', '4'],
};

export const freeFormMultiMessageErrorMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['4', '5', '6'],
};
