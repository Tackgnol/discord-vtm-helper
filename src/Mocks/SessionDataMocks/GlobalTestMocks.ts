import { GlobalTest } from '../../Models/GameData';

export const globalTestWithoutShortCircuitMock: GlobalTest = {
	testMessage: 'test',
	shortCircuit: false,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'test',
	replyPrefix: 'testPrefix',
};
export const globalTestWithShortCircuitMock: GlobalTest = {
	testMessage: 'shortCircuitTest',
	shortCircuit: true,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'scTest',
	replyPrefix: 'testPrefix',
};
