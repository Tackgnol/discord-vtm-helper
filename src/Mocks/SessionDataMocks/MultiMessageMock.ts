import { IMultiPlayerMessage } from '../../Models/GameData';

export const multiMessageMock: IMultiPlayerMessage = {
	userList: ['0'],
	name: 'test',
	message: "you've got mail",
};
