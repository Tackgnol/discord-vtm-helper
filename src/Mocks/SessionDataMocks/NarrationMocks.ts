import { INarration } from '../../Models/GameData';

export const basicNarration: INarration = {
	name: 'test',
	image: 'urlToImage',
	narrationText: 'this is a narration text',
};
