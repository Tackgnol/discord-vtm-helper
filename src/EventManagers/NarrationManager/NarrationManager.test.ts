import NarrationManager from './NarrationManager';
import { basicNarration } from '../../Mocks/SessionDataMocks/NarrationMocks';
import { expect } from 'chai';

describe('EventManagers >> NarrationManager >> NarrationManager', () => {
	const manager = new NarrationManager();
	const { narrationText, image } = basicNarration;

	it('generates a proper narration', () => {
		const result = manager.displayNarration(narrationText, image);
		expect(result).toMatchSnapshot();
	});

	it('throws an error if no text is provided', () => {
		expect(() => {
			manager.displayNarration('', image);
		}).to.throw;
	});
});
