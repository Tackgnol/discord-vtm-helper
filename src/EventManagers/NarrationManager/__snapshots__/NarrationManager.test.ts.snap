// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventManagers >> NarrationManager >> NarrationManager generates a proper narration 1`] = `
Object {
  "type": 1,
  "value": Object {
    "author": null,
    "color": 9044739,
    "description": "this is a narration text",
    "fields": Array [],
    "footer": null,
    "image": Object {
      "url": "urlToImage",
    },
    "thumbnail": null,
    "timestamp": null,
    "title": "Silence, as I show what thy must see!",
    "type": "rich",
    "url": null,
  },
}
`;
